src/main/java/org/broadinstitute/hellbender/tools/genomicsdb/GenomicsDBImport.java

```diff
@@ -221,24 +263,176 @@ public void onTraversalStart() {
      */
     @Override
     public void traverse() {
-        try {
-            importer.importBatch();
-        } catch (IOException e) {
-            throw new UserException("GenomicsDB importBatch raised exception", e);
+
+        int updatedBatchSize = (batchSize == DEFAULT_ZERO_BATCH_SIZE) ? sampleNames.size() : batchSize;
+        int sampleCount = sampleNames.size();
+        int totalBatchCount = (sampleCount/updatedBatchSize) + (sampleCount%updatedBatchSize==0 ? 0 : 1);
+
+        GenomicsDBImporter importer;
+
+        for (int i = 0, batchCount = 1; i < sampleCount; i += updatedBatchSize, ++batchCount) {
+
+            Map<String, FeatureReader<VariantContext>> sampleToReaderMap =
+                getFeatureReaders(sampleNames, sampleNameToVcfUri, updatedBatchSize, sampleCount, i);
+
+            logger.info("Importing batch " + batchCount + " with " + sampleToReaderMap.size() + " samples");
+            final long variantContextBufferSize = vcfBufferSizePerSample * sampleToReaderMap.size();
+
+            GenomicsDBImportConfiguration.ImportConfiguration importConfiguration =
+                    createImportConfiguration(workspace, arrayName,
+                            variantContextBufferSize, segmentSize,
+                            "", "",
```

Comment:
Do we still need these parameters to `createImportConfiguration()` that we're passing in empty Strings for? This is the only call to this method, so I think we could remove these two parameters.